client/src/worldgen.rs

```diff
@@ -257,19 +264,59 @@ impl ChunkParams {
         }
     }
 
-    fn generate_road_support(&self, center: na::Vector3<f64>) -> Option<Material> {
+    fn generate_road_support(
+        &self,
+        center: na::Vector3<f64>,
+        coords: na::Vector3<u8>,
+    ) -> Option<Material> {
         let plane = -Plane::from(Side::B);
         let horizontal_distance = plane.elevation(center, self.chunk);
 
-        if horizontal_distance < 0.3 {
+        if (horizontal_distance < 0.3) && self.trussing_at(coords) {
             Some(Material::WoodPlanks)
         } else {
             None
         }
     }
 
+    /// Make a truss-shaped template
+    fn trussing_at(&self, coords: na::Vector3<u8>) -> bool {
+        //Generates planar diagonals, but corner is offset
+        let mut criteria_met = 0_u32;
+        let x = coords[0];
+        let y = coords[1];
+        let z = coords[2];
+
+        // straight lines.
+        if x == 2 * self.dimension / 3 {
+            criteria_met += 1;
+        }
+        if y == 2 * self.dimension / 3 {
+            criteria_met += 1;
+        }
+        if z == 2 * self.dimension / 3 {
+            criteria_met += 1;
+        }
+
+        //main diagonal
+        if x == y {
+            criteria_met += 1;
+        }
+        if y == z {
+            criteria_met += 1;
+        }
+        if x == z {
+            criteria_met += 1;
+        }
```

Comment:
These if statements would be more concise if they were all rewritten as something like `criteria_met += u32::from(foo == bar);`